### PR TITLE
Add versioned counterparts for VolumeBindingArgs

### LINKED ISSUE #91139: Add versioned counterparts for VolumeBindingArgs
part of https://github.com/kubernetes/kubernetes/issues/90958

also, we need to add tests in pkg/scheduler/apis/config/scheme/

/sig scheduling
/kind cleanup

### PR DESCRIPTION
**Pull request by cofyc**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
/kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

- Add versioned counterparts for VolumeBindingArgs
- Scheme tests

The second commit contains auto-generated files only.
 
**Which issue(s) this PR fixes**:

Fixes #91139 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kube-scheduler/config/v1alpha2/types.go [hunk 1]
```diff
@@ -77,6 +77,8 @@ type KubeSchedulerConfiguration struct {
 	// Duration to wait for a binding operation to complete before timing out
 	// Value must be non-negative integer. The value zero indicates no waiting.
 	// If this value is nil, the default value will be used.
+	// DEPRECATED: BindTimeoutSeconds in deprecated. To change volume bind
```
Comment (ahg-g) [inline staging/src/k8s.io/kube-scheduler/config/v1alpha2/types.go:80]:
```suggestion
	// DEPRECATED: BindTimeoutSeconds is deprecated. To change volume bind
```

**Review by ahg-g — COMMENTED** [review]
looks ok to me, one nit, will leave it to Aldo to approve/lgtm

File: staging/src/k8s.io/kube-scheduler/config/v1alpha2/types.go (hunk 1 above)
Comment (cofyc) [reply staging/src/k8s.io/kube-scheduler/config/v1alpha2/types.go]: thanks! fixed

File: pkg/scheduler/apis/config/scheme/scheme_test.go [hunk 2]
```diff
@@ -66,6 +66,9 @@ profiles:
   - name: ServiceAffinity
     args:
       affinityLabels: ["bar"]
+  - name: VolumeBinding
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/scheme/scheme_test.go:69]: Add one case to "empty and no plugin args". Basically to test default values.

Comment (alculquicondor) [reply pkg/scheduler/apis/config/scheme/scheme_test.go:69]: Also there are tests in `pkg/scheduler/framework/v1alpha1/framework_test.go` `TestNewFrameworkPluginDefaults`

Comment (cofyc) [reply pkg/scheduler/apis/config/scheme/scheme_test.go:81]: updated, thanks!

**Comment by alculquicondor** [comment]
/lgtm
/approve
for scheduler

/assign @liggitt

File: staging/src/k8s.io/kube-scheduler/config/v1alpha2/types_pluginargs.go [hunk 3]
```diff
@@ -173,3 +173,13 @@ type ServiceAffinityArgs struct {
...
 	AntiAffinityLabelsPreference []string `json:"antiAffinityLabelsPreference,omitempty"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeBindingArgs holds arguments used to configure the VolumeBinding plugin.
+type VolumeBindingArgs struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// BindTimeoutSeconds is the timeout in seconds in volume binding.
```
Comment (liggitt) [inline staging/src/k8s.io/kube-scheduler/config/v1alpha2/types_pluginargs.go:183]: document the default

Comment (liggitt) [reply staging/src/k8s.io/kube-scheduler/config/v1alpha2/types_pluginargs.go:183]: does the same doc as BindTimeoutSeconds apply?

> Duration to wait for a binding operation to complete before timing out.
> Value must be non-negative integer. The value zero indicates no waiting.

**Comment by liggitt** [comment]
mechanics look good, just one question about docs

File: staging/src/k8s.io/kube-scheduler/config/v1alpha2/types_pluginargs.go (hunk 3 above)
Comment (cofyc) [reply staging/src/k8s.io/kube-scheduler/config/v1alpha2/types_pluginargs.go]: no accurate, it's better to be clear that 

- the unit is `second`
- use `volume` before `binding` to distinguish it from pod `binding` operation in the scheduler

Comment (cofyc) [reply staging/src/k8s.io/kube-scheduler/config/v1alpha2/types_pluginargs.go]: docs about the requirement, special value (`0`) and the default have been updated

**Comment by liggitt** [comment]
/approve

for API mechanics

File: staging/src/k8s.io/kube-scheduler/config/v1alpha2/types_pluginargs.go [hunk 4]
```diff
@@ -173,3 +173,15 @@ type ServiceAffinityArgs struct {
...
 	AntiAffinityLabelsPreference []string `json:"antiAffinityLabelsPreference,omitempty"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeBindingArgs holds arguments used to configure the VolumeBinding plugin.
+type VolumeBindingArgs struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// BindTimeoutSeconds is the timeout in seconds in volume binding operation.
```
Comment (alculquicondor) [inline staging/src/k8s.io/kube-scheduler/config/v1alpha2/types_pluginargs.go:183]: Copy this to the internal type `pkg/scheduler/apis/config`, except for the exact value.

Comment (cofyc) [reply staging/src/k8s.io/kube-scheduler/config/v1alpha2/types_pluginargs.go:183]: done

**Comment by alculquicondor** [comment]
This needs a rebase now :(

**Comment by alculquicondor** [comment]
Side note: we have broken autoscaler with the parent PR

https://github.com/kubernetes/kubernetes/pull/83726/files#diff-610abd3ea1ccc7b357ef99238e6803cbR176-R178

This PR should fix it, as it puts in the machinery that sets default values.

**Comment by cofyc** [comment]
@alculquicondor updated

File: staging/src/k8s.io/kube-scheduler/config/v1alpha2/types_pluginargs.go (hunk 4 above)
Comment (alculquicondor) [reply staging/src/k8s.io/kube-scheduler/config/v1alpha2/types_pluginargs.go:183]: What about `pkg/scheduler/apis/config/types_pluginargs.go`?

Comment (cofyc) [reply staging/src/k8s.io/kube-scheduler/config/v1alpha2/types_pluginargs.go:183]: my bad, updated now :(

**Comment by alculquicondor** [comment]
/lgtm
Thanks!

**Pull request merged**